scipy/stats/_sensitivity_analysis.py

```diff
@@ -575,25 +616,35 @@ def sobol_indices(
             inspect.Parameter('f_AB', kind=kind)
         )
         sig = inspect.Signature(parameters=parameters)
-        indices_method = method
+        indices_method_ = method
 
-        if inspect.signature(indices_method) != sig:
+        if inspect.signature(indices_method_) != sig:
```

Comment:
As you know, I think this needs to be changed so that the comparison doesn't fail when there are type hints.